src/steamship/data/tags/tag.py

```diff
@@ -15,17 +15,40 @@ class TagQueryRequest(Request):
 
 
 class Tag(CamelModel):
+    # Steamship client.
     client: Client = Field(None, exclude=True)
+
+    # ID of the tag in the database.
     id: str = None
+
+    # ID of the file associated with the tag.
     file_id: str = None
-    block_id: str = None
+
+    # ID of the block associated with the tag. If not None, `start_idx` and `end_idx` should be set.
+    block_id: Optional[str] = None
+
+    # The kind of tag. See the ``TagKind`` enum class for suggestions.
     kind: str = None  # E.g. ner
-    name: str = None  # E.g. person
-    value: Dict[str, Any] = None  # JSON Metadata
-    start_idx: int = None  # w/r/t block.text. None means 0 if blockId is not None
-    end_idx: int = None  # w/r/t block.text. None means -1 if blockId is not None
+
+    # The name of tag. See the ``DocTag``, ``TokenTag``, etc enum classes for suggestions.
+    name: Optional[str] = None  # E.g. person
+
+    # The value payload of the tag. Always a JSON-style object.
+    value: Optional[Dict[str, Any]] = None
+
+    # Start-index of the text covered by the tag, using python slice semantics.
+    start_idx: Optional[int] = None
+
+    # End-index of the text covered by the tag, using python slice semantics.
+    end_idx: Optional[int] = None
+
+    # The text covered by the tag. Note that text will not always be materialized into the tag object
+    # itself; you may have to fetch it with file.text[tag.start_idx:tag.end_idx]
+    text: Optional[str] = None
```

Comment:
You may also want to note that setting /overwriting this field will have no effect.

Reply:
What if we make text a read-only property that cannot be set by the user? 

That would avoid the weird case where the user ads text that does not match the highlighted span. 

For efficiency we could instruct the engine to include or exclude the text field when returning tags.

Reply:
I think a read-only property might work really well for that. Going to wait on that since the PRs after this one are going to use the text field heavily & that might provide some design information too